### PR TITLE
Search form

### PR DESCRIPTION
**Pull request by jmbarne3**
**Description**
Adds an old fashion form for searching locations, implemented via shortcode on pages. The `[location-search]` will add the form and results to any page it's added to.

**Motivation and Context**
This allows us to have a simpler solution for searching and also allows for a query param to be passed in, which we will use to redirect traffic from the current map search page to the locations page.

**How Has This Been Tested?**
Changes are available for review in dev.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [ ] My change requires an update to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: shortcodes/class-ucf-location-search-form.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	public static function callback( $atts, $content = '' ) {
+		$q = isset( $_GET['q'] ) ? sanitize_text_field( $_GET['q'] ) : null;
+
+		$atts = shortcode_atts(
+			array(
+				'post_type'      => 'location',
+				'order'          => 'ASC',
+				'order_by'       => 'post_title',
+				'posts_per_page' => -1,
+				'posts_per_row'  => 2
```
Comment (cjg89) [inline shortcodes/class-ucf-location-search-form.php:36]: Unused attr can be removed

File: shortcodes/class-ucf-location-search-form.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		if ( ! empty( $q ) ) $atts['s'] = $q;
+
+		$posts = get_posts( $atts );
+
+		ob_start();
+	?>
+		<form action="." type="get">
+			<div class="input-group mb-4">
+				<label class="sr-only" for="location-search">Search for UCF locations</label>
+				<input type="text" class="search-query form-control" id="location-search" name="q" placeholder="Library" aria-label="Search UCF Locations"<?php echo ( ! empty( $q ) ) ? ' value="' . $q . '"' : '';?>>
```
Comment (cjg89) [inline shortcodes/class-ucf-location-search-form.php:51]: Can we change the default placeholder attr to something other than "Library"--e.g. "Search Locations"/"Search locations by name"? This has thrown me off more than once/made me think I'd already submitted a search query when I hadn't

File: shortcodes/class-ucf-location-search-form.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+				<span class="input-group-btn">
+					<button class="btn btn-primary" type="submit">
+						<span class="fa fa-search" aria-labelledby="search-btn-text"></span>
+						<span id="search-btn-text" class="hidden-sm-down">Search</span>
+					</button>
+				</span>
+			</div>
+		</form>
+		<div class="location-list">
```
Comment (cjg89) [inline shortcodes/class-ucf-location-search-form.php:60]: A "X results found for [query]" blurb above the results when a search query has been submitted would be helpful here for context and for screenreader users.

Comment (cjg89) [reply shortcodes/class-ucf-location-search-form.php:60]: May also want to consider a "no search results found" blurb if nothing matched the provided query.

File: shortcodes/class-ucf-location-search-form.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+							<div class="row">
+								<?php if ( $address ) : ?>
+								<div class="col-md-7 flex-last">
+									<span class="d-block text-uppercase text-muted font-weight-light small">Address</span>
+									<p><?php echo $address; ?></p>
+								</div> <!-- End address block -->
+								<?php endif; ?>
+								<?php if ( $campus ) : ?>
+								<div class="col-md-5 flex-first">
+									<span class="d-block text-uppercase text-mited font-weight-light small">Campus</span>
```
Comment (cjg89) [inline shortcodes/class-ucf-location-search-form.php:80]: Typo `text-mited`.

`font-weight-light` probably isn't needed after making this change on this or the address span.

**Review by cjg89 — CHANGES_REQUESTED** [review]

**Review by cjg89 — APPROVED** [review]

**Review by cadie — APPROVED** [review]

**Review by rjucf — APPROVED** [review]

**Pull request merged**